brainiak/fcma/classifier.py

```diff
@@ -350,63 +377,118 @@ def fit(self, X, y, num_training_samples=None):
-        self.num_samples_ = num_samples
         if not (isinstance(self.clf, sklearn.svm.SVC)
                 and self.clf.kernel == 'precomputed'):
-            if num_training_samples is not None:
-                num_training_samples = None
-                logger.warn(
-                    'num_training_samples should not be set for classifiers '
-                    'other than SVM with precomputed kernels'
-                )
             # correlation computation
-            corr_data = self._prepare_auto_corerelation_data(X)
+            corr_data = self._prepare_corerelation_data(x1, x2)
             # normalization
             normalized_corr_data = self._normalize_correlation_data(
                 corr_data,
                 self.epochs_per_subj)
             # training data prepare
-            data = normalized_corr_data.reshape(num_samples,
-                                                num_voxels * num_voxels)
+            data = normalized_corr_data.reshape(self.num_samples_,
+                                                self.num_features_)
             self.training_data_ = None
         else:  # SVM with precomputed kernel
-            if self.num_processed_voxels < num_voxels:
+            if self.num_processed_voxels < self.num_voxels_:
                 if num_training_samples is None:
                     raise RuntimeError('the kernel matrix will be '
                                        'computed portion by portion, '
                                        'the test samples must be predefined '
                                        'by specifying '
                                        'num_training_samples')
-                if num_training_samples >= num_samples:
+                if num_training_samples >= self.num_samples_:
                     raise ValueError('the number of training samples '
                                      'must be smaller than '
                                      'the number of total samples')
             data, normalized_corr_data = \
-                self._compute_kernel_matrix_in_portion(X)
-            if self.num_processed_voxels >= num_voxels:
+                self._compute_kernel_matrix_in_portion(x1, x2)
+            if self.num_processed_voxels >= self.num_voxels_:
                 # training data is in shape
                 # [num_samples, num_voxels * num_voxels]
                 self.training_data_ = normalized_corr_data.reshape(
-                    num_samples,
-                    num_voxels * num_voxels)
+                    self.num_samples_,
+                    self.num_features_)
             else:
                 # do not store training data because it was partially computed
                 self.training_data_ = None
             logger.debug(
                 'kernel computation done'
             )
+        return data
+
+    def fit(self, X, y, num_training_samples=None):
+        """ use correlation data to train a model
```

Comment:
Actually, you missed the first line:
`"""Use correlation data to train a model.`